cg/apps/crunchy.py

```diff
@@ -0,0 +1,320 @@
+"""
+    Module for compressing BAM to CRAM
+"""
+
+import logging
+import tempfile
+from pathlib import Path
+
+from cg.utils import Process
+from .constants import (
+    BAM_SUFFIX,
+    BAM_INDEX_SUFFIX,
+    CRAM_SUFFIX,
+    CRAM_INDEX_SUFFIX,
+    FASTQ_FIRST_READ_SUFFIX,
+    FASTQ_SECOND_READ_SUFFIX,
+    SPRING_SUFFIX,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+SBATCH_HEADER_TEMPLATE = """
+#!/bin/bash
+#SBATCH --job-name={job_name}
+#SBATCH --account={account}
+#SBATCH --ntasks={ntasks}
+#SBATCH --mem={mem}G
+#SBATCH --error={log_dir}/{job_name}.stderr
+#SBATCH --output={log_dir}/{job_name}.stdout
+#SBATCH --mail-type=FAIL
+#SBATCH --mail-user={mail_user}
+#SBATCH --time=4:00:00
+#SBATCH --qos=low
+
+set -e
+
+echo "Running on: $(hostname)"
+
+source activate {conda_env}
+
+
+"""
+
+SBATCH_BAM_TO_CRAM = """
+
+error() {{
+    if [[ -e {cram_path} ]]
+    then
+        rm {cram_path}
+    fi
+
+    if [[ -e {cram_path}.crai ]]
+    then
+        rm {cram_path}
+    fi
+
+    exit 1
+}}
+
+trap error ERR
+
+crunchy --reference {reference_path} compress bam --bam-path {bam_path} --cram-path {cram_path}
```

Comment:
@moonso does this look alright? Do we want to run this with threads?

Reply:
I think we should use `--threads 12` as we discussed last week.